Downloader.py

```diff
@@ -0,0 +1,108 @@
+        return
+
+    database_file = open(database_path, 'r+')
+    temp_file = open(temp_path, 'r')
+    updated_file = open("updated", 'w+')
+    updated_file.truncate()
+
+    database = []
+    temp = []
+
+    #Read data and put into two arrays
+    for line in database_file:
+        split = line.split(',')
+
+        row = []
+        row.append(split[0])
+        row.append(split[1])
+        row.append(split[2].strip())
+
+        database.append(row)
+
+    for line in temp_file:
+        split = line.split(',')
+
+        row = []
+        row.append(split[0])
+        row.append(split[1])
+        row.append(split[2].strip())
+
+        temp.append(row)
+
+    if not len(database) == len(temp):
+        print "Database and Temp database do not match"
+
+        #Compare version numbers between temp and the database
+    for row1,row2 in zip(database,temp):
+        #Download file if needed
+        if row1[1] < row2[1]:
+            row1[1] = row2[1]
+
+            #Download File
+            print "downloading"
+            urllib.urlretrieve (row2[2], row1[2] +".zip")
+
+            #unzip file
+            print "unzipping"
+            zip_ref = zipfile.ZipFile(row1[2]+ ".zip", 'r')
+            zip_ref.extractall(row1[2])
+            zip_ref.close()
+
+            #Tile map
+            print "Creating tiles"
+            directory = row1[2]
+            filename = row1[2]
+            for files in os.listdir(directory):
+                if files.endswith(".tif"):
+                    os.rename(directory + "/" + files, directory + "/" + filename + ".tif")
+                    call(["gdal_translate", "-of","vrt","-expand", "rgba",  directory + "/" + filename +  ".tif",  directory + "/temp.vrt"])
+                    call(["gdal2tiles.py","-z","4-6", "-p", "raster", directory + "/temp.vrt", directory + "/" + filename])
+
+            #Move tfw file into tiles folder
+            for files in os.listdir(directory):
+                if files.endswith(".tfw"):
+                    os.rename(directory + "/" + files, directory + "/" + directory + "/" + filename + ".tfw")
+
+            #Zip and move folder
+            print "ziping tiles and moving files"
+            zf = zipfile.ZipFile(filename + ".zip", "w")
+            for dirname, subdirs, files in os.walk(directory + "/" + directory):
+                zf.write(dirname)
+                for filename in files:
+                    zf.write(os.path.join(dirname,filename))
+            zf.close()
+
+            os.rename(row1[2] + ".zip", chart_path + row1[2] + "/" + row1[2] + ".zip")
+
+            #clean up
+            print "cleaning up"
+            shutil.rmtree(directory)
+            did_update = True
+
+        updated_file.write(row1[0] + ',' + row1[1] + ',' + row1[2] + '\n')
+
+    if did_update:
+        print "updating database"
+        #os.rename("updated", database_path)
+    else:
+        print "no charts to update"
+
+downloader()
```

Comment:
When is this script run? And what's the flow of our system? I'm kind of gathering it is something like:
1) web scraper runs every day or something.
2) generates a temporary database
3) This downloader is script is executed XX hours/days/? and by who to update our server's database.
4) Our server disposes it's in-memory database every XX hours/days/? and reads in the updated db.
5) App gets new info.

Is that what you were thinking?

I'm worried about who owns this downloader process. Maybe the flow should be like
1) web scraper runs, generates temp db with all necessary model info.
2) web scraper runs this downloader script to compare the server's db to its own and download/tile new versions. 
3) I think what Clauson was saying is good: Have this downloader generate a new folder that contains (a) a file that holds all the information needed for the model (version number, dates, etc) and (b) the zipped folder of tiles.
4) Then the server notices that a folder has been added or changed (maybe with something like [this](https://pypi.python.org/pypi/watchdog)) and then opens up the file storing the model information, and replaces that line in its db with the new info. Or the web scraper could hit a route that tells the server to update its db, which would induce a directory scan to look for new directories or something.